src/mame/nmk/nmk214.cpp

```diff
@@ -0,0 +1,191 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergio Galiano
+/*
+    NMK214 GFX Descrambler emulation
+	
+	This device is used for descrambling the GFX data on some game pcbs from NMK (nmk16).
+	It works in tandem with NMK215, that is a Toshiba MCU which sends initialization data to NMK214 to perform the descrambling process.
+	Every game pcb using it has two units of NMK214, one for descrambling sprite data and another one for descrambling background tiles data.
+	It can work in two different modes: word and byte. For sprites it always works in word mode. On the order side, for backgrounds it always works in byte mode.
+	There are 8 different internal configs hardwired inside the device, and the data received from NMK215 selects one of those at startup. That init data is stored in the device when bit 3 matches with the operation mode wired directly on the pcb.
+	The descrambling process is essentially a dynamic bitswap of the incoming word/byte data, doing a different bitswap based on the address of the data to be descrambled.
+	The input address bus on the device is used to determine which bitswap do for each word/byte, and it's usually hooked up in different way for sprites and bg tiles, so an 'input_address_bitswap' is included in the implementation in order to get the effective address the device will use internally for each case.
```

Comment:
Please wrap these lines at a reasonable width.

Reply:
Just an FYI, I use a wrap width of 140 characters in my Guru-Readmes(tm)(c)(r). This allows everything to fit on a laptop screen that was common years ago and used on current low-end laptops (i.e. very common WXGA 1280x800)

Reply:
Fixed